Add file encryption using box and the enclave's public key

longtomjr opened the pull request:
closes: #8

longtomjr committed:
feat(upload): move state logic to vuex and add decrypt

longtomjr committed:
feat(upload): add components to show upload result

jongbonga commented on src/views/Home.vue:
```diff
@@ -7,7 +7,7 @@
               <span>JWT Token</span>
               <span>
-                <el-button size="small" @click="fetchJWT()"> Fetch </el-button>
+                <el-button size="small" @click="getAttestation()"> Fetch </el-button>
```
@longtomjr should we remove the "validate" button that was previously getting the attestation?

jongbonga committed:
remove unused button & declaration

PiDelport commented:
@jongbonga: It looks like the merge accidentally undid commit c3f0317… I'll push a rebase that includes that, and we can go over it?

PiDelport committed:
style(components/FileUpload): yarn lint

PiDelport committed:
style: unused imports

PiDelport committed:
style(store): unused parameters

PiDelport committed:
refactor(store): log placeholder functionality to console

(This makes the TODOs executable, and uses the variables for eslint.)

PiDelport committed:
refactor: tweak console logging for readability

PiDelport committed:
fix(FileUpload): add missing Boolean type coercion

PiDelport committed:
refactor(cryptography): drop unused encryptSecretbox

(in favour of encryptBox)

PiDelport committed:
refactor: naming

We should probably try to be consistent about using "their" / "our" for
naming public / secret key variables, rather than encoding a role name
like "enclave" for one half of the exchange.

(These generic helper functions may be used to encrypt data for roles
other than the enclave.)

PiDelport committed:
fix(FileUpload): add missing await

PiDelport committed:
fix(Home): add missing await

PiDelport reviewed (commented):
Looks good!

I added commits for the issues I spotted: can you check that it looks good?

longtomjr commented on src/components/FileUpload.vue:
```diff
@@ -37,6 +39,12 @@
           <el-descriptions-item label="Extension">
             {{ state._file.extension }}
           </el-descriptions-item>
+          <el-descriptions-item label="Upload password">
```
Rename "password" to "access-key"

longtomjr committed:
feat(upload): rename password to accessKey
